Packs/EDL/Integrations/EDL/EDL.py

```diff
@@ -486,22 +530,37 @@ def route_edl_values() -> Response:
     credentials = params.get('credentials') if params.get('credentials') else {}
     username: str = credentials.get('identifier', '')
     password: str = credentials.get('password', '')
+    cache_refresh_rate: str = params.get('cache_refresh_rate')
     if username and password:
         headers: dict = cast(Dict[Any, Any], request.headers)
         if not validate_basic_authentication(headers, username, password):
             err_msg: str = 'Basic authentication failed. Make sure you are using the right credentials.'
             demisto.debug(err_msg)
-            return Response(err_msg, status=401)
+            return Response(err_msg, status=401, mimetype='text/plain', headers=[
+                ('WWW-Authenticate', 'Basic realm="Login Required"'),
+            ])
 
     request_args = get_request_args(request.args, params)
     on_demand = params.get('on_demand')
-
+    created = datetime.now(timezone.utc)
     values = get_edl_ioc_values(
         on_demand=on_demand,
         request_args=request_args,
-        cache_refresh_rate=params.get('cache_refresh_rate'),
+        cache_refresh_rate=cache_refresh_rate,
     )
-    return Response(values, status=200, mimetype='text/plain')
+    query_time = (datetime.now(timezone.utc) - created).total_seconds()
+    edl_size = values.count('\n') + 1  # add 1 as last line doesn't have a \n
+    max_age = ceil((datetime.now() - dateparser.parse(cache_refresh_rate)).total_seconds())
+    demisto.debug(f'Returning edl of size: [{edl_size}], created: [{created}], query time seconds: [{query_time}],'
```

Comment:
@glicht `demisto.debug` is not thread-safe. I'm wondering if it should be protected with a lock. This comment should also be true for the handlers using `demisto.debug` and `demisto.info`.

Reply:
It's ok. As we use gevent we are single threaded.